doc: Improvements around Code of Conduct

miri64 opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md.
-->

### Contribution description
This moves

- Some wiki pages related to our code of conduct (reporting guidelines and FAQ) to our documentation,
- Adds the Code of Conduct itself to the documentation (as a copy of [CODE_OF_CONDUCT.md](https://github.com/RIOT-OS/RIOT/blob/master/CODE_OF_CONDUCT.md)), and
- Provides an overview over all of these pages as a new “Code of Conduct Information” page. The main purpose of the latter is so all of them are collected under one group in the Doxygen side bar.
<!--
Put here the description of your contribution:
- describe which part(s) of RIOT is (are) involved
- if it's a bug fix, describe the bug that it solves and how it is solved
- you can also give more information to reviewers about how to test your changes
-->


### Testing procedure
Read the results generated by Murdock.
<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references
None
<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

mguetschow commented on doc/doxygen/src/coc-faq.md:
```diff
@@ -0,0 +1,72 @@
+# FAQ  {#coc-faq}
+
+This FAQ attempts to address common questions and concerns around the RIOT
+community's future [Code of Conduct][CoC]. If you still have questions after
```
```suggestion
community's [Code of Conduct][CoC]. If you still have questions after
```

Or we leave it as is to be a perfect copy of the Wiki page and change it in a follow-up PR.

miri64 replied:
In the past, I updated it just with a follow-up commit. So I would just do it here.

mguetschow reviewed (approved):
Checked the generated documentation at this deeplink: https://ci.riot-os.org/results/4efbd5afc4e74cb69f98212986e6ff8b/doc-preview/coc-info.html

Looks good to me, but maybe we want to have more than one ACK on this one.

miri64 commented:
> Looks good to me, but maybe we want to have more than one ACK on this one.

This PR just moves files around. Do we really need more ACKs for this?

mguetschow commented:
Thanks @miri64 !

mguetschow commented:
Should we go on and remove the Wiki pages (or rather link to the documentation pages instead)?

miri64 commented:
> Should we go on and remove the Wiki pages (or rather link to the documentation pages instead)?

Done in the same style I removed all the other pages I moved to the doc in the past few weeks: https://github.com/RIOT-OS/RIOT/wiki/_history
